Tests/AsyncAlgorithmsTests/TestValidationTests.swift

```diff
@@ -12,6 +12,7 @@
 import XCTest
 import AsyncAlgorithms
 import AsyncSequenceValidation
+@testable import AsyncAlgorithms_XCTest
```

Comment:
so this prevents us from doing release-mode testing of perf. I am of the belief that this DEFINITELY should be tested, but we kinda need to settle on a way to do this AND accommodate for testing performance too

Reply:
I actually didn't realize that this prevents that. I thought about moving the tests to their own test target for the new module, but they depend on other types in the `AsyncAlgorithmsTests` module and I tried to make as few changes as possible. I guess making more of the methods in the extension doesn't feel like a good solution either?